content/en/docs/instrumentation/ruby/manual.md

```diff
@@ -8,18 +8,70 @@ aliases:
 weight: 4
 ---
 
-Auto-instrumentation is the easiest way to get started with instrumenting your code, but in order to get the most insight into your system, you should add manual instrumentation where appropriate.
-To do this, use the OpenTelemetry SDK to access the currently executing span and add attributes to it, and/or to create new spans.
+Auto-instrumentation is the easiest way to get started with instrumenting your
+code, but in order to get the most insight into your system, you should add
+manual instrumentation where appropriate. To do this, use the OpenTelemetry SDK
+to access the currently executing span and add attributes to it, and/or to
+create new spans.
 
-### Add information to the current span
+## Initializing the SDK
 
-It's often beneficial to add information to the currently executing span in a trace.
-For example, you may have an application or service that handles extended warranties, and you want to associate it with the span when querying your tracing datastore.
-In order to do this, get the current span and set [attributes](#attributes) with your application's domain specific data:
+First, ensure you have the SDK package installed:
+
+```console
+gem install opentelemetry-sdk
+```
+
+Then include configuration code that runs when your program initializes. Make
+sure that `service.name` is set by configuring a service name.
+
+### Acquiring a Tracer
+
+To being [tracing](/docs/concepts/signals/traces), you will need to ensure you
+have an initialized [`Tracer`](/docs/concepts/signals/traces#tracer) that comes
+from a [`TracerProvider`](/docs/concepts/signals/traces#tracer-provider).
+
+The easiest and most common way to do this is to use the globally-registered
+TracerProvider. If you are using [instrumentation
+libraries](/docs/instrumentation/ruby/automatic), such as in a Rails app, then
+one will be registered for you.
 
 ```ruby
+# If in a rails app, this lives in config/initializers/opentelemetry.rb
+require "opentelemetry/sdk"
+
+begin
+  OpenTelemetry::SDK.configure do |c|
+    c.service_name = '<YOUR_SERVICE_NAME>'
+  end
+rescue OpenTelemetry::SDK::ConfigurationError => e
+  puts "Error configuring Ruby OpenTelemetry SDK"
+  puts e.inspect
+end
```

Comment:
So, technically this example means that someone could have an incorrectly or uninitialized tracing SDK. If we advised people to rescue configuration errors _in general_ this is fine, but otherwise I might suggest not rescuing the error.

Reply:
Yeah, good point. I'll remove this for now. This is likely a topic we'd need to address for all languages at some point, and it's unlikely that what I have here is exactly the right best practice in general.